### ISSUE TITLE
QUIC: dereference NULL pointer in function qtx_reserve_txe()

### ISSUE THREAD

**Issue by nv-dmd** [openssl/openssl#23612]
The function ossl_qtx_write_pkt() (ssl/quic/quic_record_tx.c) has a call qtx_reserve_txe(qtx, NULL, txe, qtx->mdpl) on line 827
Call stack
qtx_reserve_txe(qtx, NULL, txe, qtx->mdpl) -> qtx_resize_txe(qtx, NULL, txe, n) -> ossl_list_txe_remove(NULL, txe);
This looks like there will be a dereference NULL in the function ossl_list_txe_remove(txl, txe) where txl equal NULL

Can you please confirm or deny this?

**Comment by t8m** [comment]
Yes, you're right. Fortunately qtx_resize_txe() will never be called with the current code as the qtx->mdpl never changes from the initial value. However we should still handle the case where txl is NULL correctly. Or, maybe we could simplify the qtx_resize_txe() code to assert if the txe is on any list, i.e., allowing only unlisted txe to be resized. @hlandau might have an opinion whether resizing txes that are placed on a list will ever be necessary.

**Comment by avoget** [comment]
Hello Tomas.

Calling the **qtx_reserve_txe** method seems redundant.
Is there a need to call this?
Why not remove the call 
**qtx_reserve_txe** and the methods themselves **qtx_reserve_txe, qtx_resize_txe** because they do not perform useful actions?
